### ISSUE TITLE
Tile windows: Hide active hint when window is the only one in a workspace

### ISSUE THREAD

**Issue by aral** [pop-os/shell#280]
**Distribution (run `cat /etc/os-release`):**

NAME="Pop!_OS"
VERSION="20.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
LOGO=distributor-logo-pop-os
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
n/a

**Issue/Bug Description:**

The active hint displays for a window even if the window is the only one in a workspace. 

**Expected behavior:**

The active hint should not display for a window if the window is the only one in a workspace as it is not necessary.

**Comment by mmstick** [comment]
I'm not sure if we want to have special behaviors like this. There was a thought to disable gaps when there is only a single window, but it was decided that having the UX change based on the number of windows in a tree would make the UX inconsistent.

**Comment by aral** [comment]
I agree that having gaps disappear for a single window would be inconsistent (and as you pointed out in the other issue, would be confused with a maximized window).

On this though, it’s more about affordances. A single window is clearly the active area. It doesn’t need an additional affordance. Similarly, a modal dialogue is also clearly the active area as it appears above the other windows.

Where we lose that affordable is when windows are next to each other. There is nothing to tell you which is the active one. In that case, you need the indicator.

So I would argue that if a convention of indicating only when not obvious would have its own internal consistency and reduce noise when unnecessary.

**Comment by iclanzan** [comment]
> A single window is clearly the active area.

Not if focus is on the top bar.

**Comment by aral** [comment]
Ah interesting. So the focus indicator is meant to act as an accessibility feature and highlight every active element? If I focus, say, the main volume/WiFi/etc popover, it does not highlight it and remains on the main window.

If it’s an accessibility highlight, I would expect it to highlight that popover too. If it’s just for windows, I would expect it not to highlight a single window.

**Comment by iclanzan** [comment]
Yeah, you are right. I was expecting the active hint on the window to be removed when focusing the top bar with `Ctrl+Alt+Esc` but that doesn’t happen :man_shrugging:

**Comment by jmmaranan** [comment]
The active-hint is only focusing on the current opened window list. While it isn't probably difficult to tag a single window to not display active-hint, I think this would be best decided by the UX team for consistency now that smart gaps had been merged too.

**Comment by elshize** [comment]
> I'm not sure if we want to have special behaviors like this. There was a thought to disable gaps when there is only a single window, but it was decided that having the UX change based on the number of windows in a tree would make the UX inconsistent.

Hi @mmstick I'm a little late to the discussion, but I figure I'd add my few cents. As I'm writing this, pop-shell has "smart gaps" option that makes the only window on the desktop fill all the available space. I suppose this was not decided yet back when you wrote the above comment. Anyway, seems like you guys decided to add this option, which seems reasonable to me. Yet, the active hint is still there, even if the whole area is occupied. In my opinion, if implementing smart gaps function, it makes much more sense to skip the hint (though I agree that it makes more sense for it to be there without smart gaps). Not an UX expert, but to me it just begs for removing the border. Note that this could be also an additional option in the config.

I am used to the smart gaps from other tiling window managers but because the active hint was just weird to me, I opted to not doing smart gaps at all.

Just something to think about in the future, there might be more people like me, but I can't be sure.

**Comment by elshize** [comment]
Since I'm already here, I just want to say you're doing a great job on this extension, and your work is greatly appreciated! :tada:

**Comment by PeSader** [comment]
Maybe hiding the hint upon maximization could be an additional setting (with the same default value to behave the same as it does now).

The implementation could be similar to removing the hint on full screen, which was implemented on commit [9f0976f](https://github.com/pop-os/shell/commit/d1d2b0e0cd77a57d9f50f2b4d6199faf60d9531d).

Adding this option would allow the hints to be hidden when only one window is on the workspace, by using the Smart Gaps feature :)

**Comment by dennisschoepf** [comment]
Great work on the extension - Just wanted to know if there is any update on this?
Would second the reply of @PeSader that a toggle in the Extension Settings similar to the "Smart Gaps" could be the best option.
